### PR TITLE
Fixtures

### LINKED ISSUE #102: Add fixtures
## About

Add fixtures for easier development experience

### PR DESCRIPTION
**Pull request by andyrichardson**
## About

Adds fixtures!

Fix #102 

## Changes
 - Add fixtures (obviously)
 - Replace prop drilling w/ context
 - Replace "update flash" logic with React Spring
 - Clean up folder structures
 - Add visual regression testing
 - Speed up dependency installation in CI

## Screenshots

Kapture 2020-03-03 at 19 11 29

### CODE REVIEW

File: cosmos.override.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const webpack = require("webpack");
+
+module.exports = (c, env) => {
+  console.log(c);
```
Comment (sofiapoh) [inline cosmos.override.js]: Leftovers?

**Comment by andyrichardson** [comment]
@sofiapoh yeah I'm down for colocating data if you have suggestions on how we could do that! 

I found that most fixtures require very specific mocked states (usually unrelated to other fixtures) so I'm not sure how/if we would should merge them into one. I've gone as far as to not share any values between fixtures because deeply nested spreading hasn't been fun and its felt easier to just copy-paste/simulate real world values - but maybe there's a better way to do this!

> I also noticed you removed HighlightUpdate from the ListItem, was there a specific reason for this change?

It's back! Disabled it while trying to hook up context :+1:

File: src/panel/context/Explorer/index.tsx [hunk 2]
```diff
@@ -3,38 +3,60 @@ import React, {
...
 import { DevtoolsContext } from "../Devtools";
-import { startQuery, NodeMap } from "./ast";
+import { startQuery, NodeMap, FieldNode } from "./ast";
 
-export const ExplorerContext = createContext<NodeMap>(Object.create(null));
-
-interface Props {
-  children: ReactNode;
+export interface ExplorerContextValue {
+  operations: NodeMap;
```
Comment (andyrichardson) [inline src/panel/context/Explorer/index.tsx]: Moved explorer-wide values to context so we don't need to pass them down

File: src/panel/pages/explorer/Explorer.tsx [hunk 3]
```diff
@@ -1,38 +1,25 @@
...
-import { Tree } from "./Tree";
-import { DetailViewPane } from "./DetailView";
+import { Tree, NodeInfoPane } from "./components";
 
 export function Explorer() {
-  const operations = useContext(ExplorerContext);
-  const [focusedNodeId, setFocusedNodeId] = useState<string | undefined>(
-    undefined
-  );
-  const [detailViewNode, setDetailViewNode] = useState<FieldNode | null>(null);
```
Comment (andyrichardson) [inline src/panel/pages/explorer/Explorer.tsx]: This is now in context (focusedNode)

File: src/panel/pages/explorer/Explorer.tsx [hunk 4]
```diff
@@ -1,38 +1,25 @@
...
 import { Background } from "../../components";
 import { ExplorerContext } from "../../context";
-import { FieldNode } from "../../context/Explorer/ast";
-import { Tree } from "./Tree";
-import { DetailViewPane } from "./DetailView";
+import { Tree, NodeInfoPane } from "./components";
 
 export function Explorer() {
-  const operations = useContext(ExplorerContext);
-  const [focusedNodeId, setFocusedNodeId] = useState<string | undefined>(
```
Comment (andyrichardson) [inline src/panel/pages/explorer/Explorer.tsx]: I coudn't work out how this is any different to detailViewNode and wanted to trim down state + renders - from what I can tell, both values get set at the same time

File: src/panel/pages/explorer/Explorer.tsx [hunk 5]
```diff
@@ -49,10 +36,6 @@ const ListContainer = styled.section`
   padding-right: 1rem;
   padding-bottom: 1rem;
   overflow: auto;
-
-  @media (min-aspect-ratio: 1/1) {
```
Comment (andyrichardson) [inline src/panel/pages/explorer/Explorer.tsx]: This was probably needed before we updated the `Pane` component

File: src/panel/pages/explorer/components/Value.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    previousValue.current = value;
+  }, [value, flash, isRoot]);
+
+  const content = useMemo(() => {
+    const props = { value, expand };
+
+    if (value instanceof Array) {
+      return <ArrayValue {...props} value={value as any[]} />;
```
Comment (andyrichardson) [inline src/panel/pages/explorer/components/Value.tsx]: I've broken these conditions out and we have each different state as a fixture now

File: src/panel/pages/request/components/Response.tsx [hunk 7]
```diff
@@ -29,14 +28,16 @@ const Heading = styled.h2`
   margin: 0;
   padding: 10px;
   font-size: 12px;
-  color: ${p => p.theme.grey["+2"]};
-  background: ${props => props.theme.dark["-2"]} !important;
+  color: #fff;
+  background: ${props => props.theme.blue["0"]};
```
Comment (andyrichardson) [inline src/panel/pages/request/components/Response.tsx]: Fixed background color not changing - also fixtured

**Pull request merged**